api/prisma/createVersionedDOIs.ts

```diff
@@ -0,0 +1,120 @@
+            },
+            doi: null // only get versions without DOI
+        },
+        include: {
+            publication: {
+                select: {
+                    id: true,
+                    type: true,
+                    doi: true,
+                    topics: true,
+                    publicationFlags: true,
+                    url_slug: true
+                }
+            },
+            publicationStatus: {
+                select: {
+                    status: true,
+                    createdAt: true,
+                    id: true
+                },
+                orderBy: {
+                    createdAt: 'desc'
+                }
+            },
+            funders: {
+                select: {
+                    id: true,
+                    city: true,
+                    country: true,
+                    name: true,
+                    link: true,
+                    ror: true
+                }
+            },
+            coAuthors: {
+                select: {
+                    id: true,
+                    email: true,
+                    linkedUser: true,
+                    publicationVersionId: true,
+                    confirmedCoAuthor: true,
+                    approvalRequested: true,
+                    createdAt: true,
+                    reminderDate: true,
+                    isIndependent: true,
+                    affiliations: true,
+                    user: {
+                        select: {
+                            firstName: true,
+                            lastName: true,
+                            orcid: true
+                        }
+                    }
+                },
+                orderBy: {
+                    position: 'asc'
+                }
+            },
+            user: {
+                select: {
+                    id: true,
+                    orcid: true,
+                    firstName: true,
+                    lastName: true,
+                    email: true,
+                    createdAt: true,
+                    updatedAt: true
+                }
+            }
+        }
+    });
+
+    console.log(`Found ${latestLiveVersions.length} without a DOI.`);
+
+    let createdVersionDOIsCount = 0;
+
+    for (const version of latestLiveVersions) {
+        // create a new DOI for each version
+        console.log(`Creating version ${version.versionNumber} DOI for publication ${version.versionOf}`);
+        const versionDOIResponse = await helpers.createPublicationVersionDOI(version);
+        const versionDOI = versionDOIResponse.data.attributes.doi;
+
+        console.log(`Successfully created version ${version.versionNumber} DOI: ${versionDOI}`);
+
+        // update publication "HasVersion" with the created DOI
+        console.log(`Updating canonical DOI: ${version.publication.doi} "HasVersion" with version DOI: ${versionDOI}`);
+        await helpers.updatePublicationDOI(version.publication.doi, version);
```

Comment:
Think there is a problem here - the second parameter we pass is the version we need, but it doesn't have the doi field set yet. So when it goes to set "HasVersion" in the helper function, the value of that ends up being null.

<img width="211" alt="Screenshot 2023-11-01 131941" src="https://github.com/JiscSD/octopus/assets/132363734/29c48a6f-756f-45f3-8f98-1fd20c8e6476">

Reply:
Yeah, I remember updating that helper function after I have created and tested the script already. Fixed it now.

Reply:
Looks good